### PR TITLE
Make bootstrap.sh more friendly to other shells.

### PR DESCRIPTION
**Pull request by andy31415**
I found some CI systems trying to use dash, and that does not seem to work at all due to it not understanding `<<<` or `[[`.

Dash is still not ok because overall because arguments to `. scripts/activate.sh` do not get passed along, however it would work as a default invocation.

Changes:
  - loop using IFS (and set sh_word_split for ZSH)
  - replaced `[[` tests with `[`
  - removed restyler for this file, because it tries quoting `for platform in ${_SETUP_PLATFORM};` and that breaks the script. I found no way to tell shellharden to not try to fix that. Still checking...

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: scripts/setup/bootstrap.sh [hunk 1]
```diff
@@ -32,13 +32,14 @@ _install_additional_pip_requirements() {
...
 
-    if ! [ -z "$_SETUP_PLATFORM" ]; then
+    if [ -n "$_SETUP_PLATFORM" ]; then
+        _OLD_IFS=$IFS
+        IFS=","
         if [ -n "$ZSH_VERSION" ]; then
-            IFS="," read -r -A _PLATFORMS <<<"$_SETUP_PLATFORM"
-        else
-            IFS="," read -r -a _PLATFORMS <<<"$_SETUP_PLATFORM"
+            setopt sh_word_split
```
Comment (bzbarsky-apple) [inline scripts/setup/bootstrap.sh:39]: Wait, doesn't this need to be `setopt shwordsplit`?

Comment (andy31415) [reply scripts/setup/bootstrap.sh]: well, my change worked with this as-is and did not work without, so maybe they are aliases.

Comment (andy31415) [reply scripts/setup/bootstrap.sh]: Updated anyway as the non-underscore seems to appear more often in docs.

**Comment by andy31415** [comment]
fast tracking: infra change not functionality, tested locally on various shells, CI validates builds.

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**